### PR TITLE
trace-agent: follow-up trace normalization

### PR DESCRIPTION
**Pull request by djova**
### What does this PR do?

Follow-up to https://github.com/DataDog/datadog-agent/pull/3738

Changes:
* use clearer default names `unnamed-{language}-service` and `unnamed-operation` instead of previous names `service` and `service.trace` which could have caused confusion.
* missing start date now defaults to `now - duration`, which is a more reasonable default than simply `now` as the latter means at the time of the normalization the span is ending in the future.

### Motivation

Minor improvements upon the initial trace normalization work.

### CODE REVIEW

File: releasenotes/notes/trace-normalization-follow-up-c7a22fc4ae306f83.yaml [hunk 1]
```diff
@@ -0,0 +1,13 @@
...
+# CHANGELOG.rst is rendered. So the text needs to be worded so that
+# it does not depend on any information only available in another
+# section. This may mean repeating some details, but each section
+# must be readable independently of the other.
+#
+# Each section note must be formatted as reStructuredText.
+---
+enhancements:
+  - |
+    For trace normalization, use clearer default names `unnamed-service` and `unnamed-operation` instead of previously
```
Comment (gbbr) [inline releasenotes/notes/trace-normalization-follow-up-c7a22fc4ae306f83.yaml]: The change log only needs to reflect changes between releases, not between PRs. I think you just need to edit the old reno.

File: pkg/trace/api/normalizer.go [hunk 2]
```diff
@@ -123,6 +118,15 @@ func normalize(ts *info.TagStats, s *pb.Span) error {
 		log.Debugf("Fixing malformed trace. Duration is too large and causes overflow (reason:invalid_duration), setting span.duration=0: %s", s)
 		s.Duration = 0
 	}
+	if s.Start < Year2000NanosecTS {
```
Comment (gbbr) [inline pkg/trace/api/normalizer.go]: Can we change line 116 to use math.MaxInt64 for the overflow check?

File: releasenotes/notes/trace-normalization-follow-up-c7a22fc4ae306f83.yaml (hunk 1 above)
Comment (djova) [reply releasenotes/notes/trace-normalization-follow-up-c7a22fc4ae306f83.yaml]: Makes sense, updated

File: pkg/trace/api/normalizer.go (hunk 2 above)
Comment (djova) [reply pkg/trace/api/normalizer.go]: Yeah I guess that's a bit clear than actually causing an overflow to check it. Updated

File: pkg/trace/api/normalizer_test.go [hunk 3]
```diff
@@ -218,6 +220,17 @@ func TestNormalizeStartTooSmall(t *testing.T) {
 	assert.True(t, s.Start > before, "start should have been reset to current time")
 }
 
+func TestNormalizeStartTooSmallWithLargeDuration(t *testing.T) {
```
Comment (gbbr) [inline pkg/trace/api/normalizer_test.go]: Can we at the very least merge the `TestNormalizeStart` tests and make the rest subtests? e.g.:
```go
    t.Run("pass", ...)
    t.Run("small", ...)
    t.Run("large-duration", ...)
```

**Review by gbbr — DISMISSED** [review]
LGTM

File: pkg/trace/api/normalizer_test.go (hunk 3 above)
Comment (djova) [reply pkg/trace/api/normalizer_test.go]: Ok, updated

File: pkg/trace/api/normalizer_test.go [hunk 4]
```diff
@@ -199,23 +201,33 @@ func TestNormalizeNoSpanID(t *testing.T) {
...
+func TestNormalizeStart(t *testing.T) {
+	t.Run("pass-through", func(t *testing.T) {
+		ts := newTagStats()
+		s := newTestSpan()
+		before := s.Start
+		assert.NoError(t, normalize(ts, s))
+		assert.Equal(t, before, s.Start)
+		assert.Equal(t, newTagStats(), ts)
+	})
+	t.Run("too-small", func(t *testing.T) {
```
Comment (gbbr) [inline pkg/trace/api/normalizer_test.go]: Sorry to be nitpicking, but we need a bit of space between each tests: an empty line. It's the same everywhere.

Comment (djova) [reply pkg/trace/api/normalizer_test.go]: updated

**Review by gbbr — DISMISSED** [review]

**Review by gbbr — DISMISSED** [review]

**Review by gbbr — DISMISSED** [review]

**Review by gbbr — APPROVED** [review]

**Pull request merged**